### PR TITLE
luci-app-samba4: add new application

### PR DESCRIPTION
**Pull request by Andy2244**
Maintainer: me
Compile tested: arm/mips (master)
Run tested: arm/mvebu (master)

Description:
Its the updated luci-ui for this PR openwrt/packages#5563.
Basically just a copy from the latest `luci-app-samba`, with some added extra options for samba 4 and changed deps/names.

### CODE REVIEW

File: applications/luci-app-samba4/luasrc/model/cbi/samba4.lua [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+cm.size = 4
+
+dm = s:option(Value, "dir_mask", translate("Directory mask"))
+dm.rmempty = true
+dm.size = 4
+
+s:option(Value, "vfs_objects", translate("Vfs objects")).rmempty = true
+
+function m.on_commit(self,map)
+	require("luci.sys").call('/sbin/reload_config')
```
Comment (jow-) [inline applications/luci-app-samba4/luasrc/model/cbi/samba4.lua]: Please remove this, it should not be needed when `samba4-server` ships with a procd enabled init script.

File: applications/luci-app-samba4/luasrc/model/cbi/samba4.lua [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+tmpl.template = "cbi/tvalue"
+tmpl.rows = 20
+
+function tmpl.cfgvalue(self, section)
+	return nixio.fs.readfile("/etc/samba/smb.conf.template")
+end
+
+function tmpl.write(self, section, value)
+	value = value:gsub("\r\n?", "\n")
+	nixio.fs.writefile("//etc/samba/smb.conf.template", value)
```
Comment (jow-) [inline applications/luci-app-samba4/luasrc/model/cbi/samba4.lua]: Please remove the double slash

**Comment by Andy2244** [comment]
@jow- ok fixed

**Comment by jow-** [comment]
Merged, thank you!

**Comment by Andy2244** [comment]
@jow- Thanks, but you noticed that the parent openwrt/packages#5563 is not merged yet?

**Comment by jow-** [comment]
Sorry, didn't notice that.

**Pull request merged**